### ISSUE TITLE
Remarkable 2 devices have screen calibration issue

### ISSUE THREAD

**Issue by xdrr** [remarkableno/linux#13]
All remarkable 2 devices have a screen calibration issue causing ink to be drawn at some displacement from the nib position. The calibration offset occurs in "patches" across the screen.

Reproduction steps:

1. Procure any brand new reMarkable 2 hardware platform.
2. Obtain any EMR stylus for writing on screen including reMarkable 2 stylus (regular and pro), Lamy EMR pen, Onyx Boox pen, Samsung Galaxy pen etc.
3. With all of the above stylii write on the screen (perhaps with a ruled line template to ensure writing is straight)
4. Note at various locations across the screen area the ink is displayed at some displacement +/- 0.5mm from the position of the nib whereas at other areas the ink - correctly - appears directly below the nib.

Seeing as this affects every remarkable 2 unit, what is the plan to resolve this issue? Can this be resolved with kernel updates?

**Comment by Karlinator** [comment]
Chiming in here to both confirm the issue and add some points.

In principle it should be possible to create a calibration task ("tap the points with your stylus") to identify the areas where the tracking is offset, create a sort of offset map for the screen, and use interpolation between these points (as the offset appears to vary smoothly) to apply a correction offset to the pen event.

To avoid having to click 1000 points the calibration might start relatively coarse (36 (6x6) data points maybe) and then ask for more data surrounding points where it detected a significant offset.

Side note: https://github.com/funkey/recept/ demonstrates that it is possible to intercept the pen events and apply some magic to them even for a third-party hack (without delving into the device driver). Something like this can perhaps be used to mod this problem away if reMarkable don't fix it.

**Comment by LoganDark** [comment]
I just found this issue, and it turns out I'm already working on a project to help remedy this. Discord transcript from Jan 5th:

> My reMarkable 2 suffers from bad wacom digitizer calibration, do any of you have the same issue? I'm going to (attempt to) make a calibration utility that allows for software correction of the issue (since reMarkable failed to do it in firmware)
>
> Specifically I'm referring to inaccuracy of the marker tip, where on some parts of the display it's spot-on but in other parts it looks more like this:
>
> ![](https://cdn.discordapp.com/attachments/386181213699702786/928476810931474502/148243167-2dc6df74-eab4-45fa-bbea-acaebeec83ad.png)
>
> I'm thinking that can be helped in software by a manual calibration sequence similar to what SMART boards use, where you tap the marker tip as accurately as you can on a grid of dots, and it calculates some sort of mapping between the grid & the actual signals received from the digitizer, to correct them to be more accurate to the grid (and therefore more accurate while drawing)
>
> Image of the calibration - you tap the marker on the point shown on-screen, and the calibration assumes that where you tapped = that point that was shown, so that it can correct for it
>
> ![](https://cdn.discordapp.com/attachments/386181213699702786/928477666644996116/iu.png)
>
> Recept (https://github.com/funkey/recept) solves a similar issue, but what I'm dealing with is slightly inaccurate positioning, not jitteryness
>
> It's (supposedly) possible to load multiple different libraries using LD_PRELOAD so I'm probably going to take advantage of that for those who want to use both recept and the 'recalibration filter'
>
> You'd load the filter first and then recept of course, so recept works on the filtered data
>
> Please let me know your thoughts, if you have a device with this issue, and if you'd be down to test the program once I have a working PoC

Let me know if you'd be interested in being a tester (join the reMarkable Discord server: https://discord.gg/JSSGnFY - I'm LoganDark there)

**Comment by aogden41** [comment]
Any update on this? Pretty game breaking issue.

**Comment by istvanjaray** [comment]
@aogden41 look at this link, below.
To me, this solved it all. Just swipe the magnet of your pen over the screen thoroughly.
https://www.reddit.com/user/IoannisHere/comments/m83t6v/remarkable_2_tablet_screen_magnetization/

**Comment by aogden41** [comment]
> @aogden41 look at this link, below. To me, this solved it all. Just swipe the magnet of your pen over the screen thoroughly. https://www.reddit.com/user/IoannisHere/comments/m83t6v/remarkable_2_tablet_screen_magnetization/

Hi - I have seen this article and similar many times. This is not solving my issue, only moving the problem around the screen.

**Comment by xdrr** [comment]
@aogden41 @istvanjaray I agree with aogden41. Although the issue might be partly or wholly caused by an issue of screen magnetisation, it's not completely fixed by swiping the pen over the screen once or even each time you notice a calibration issue. There's always an issue with calibration on some part of the screen.

It would be great if someone was willing to pour ferous powder over their remarkable 2 to visualise the field lines and connect it with calibration issues.

@LoganDark  this suggestion would be a good intermediate fix if this is entirely a hardware fault. It's a shame remarkable continue to sell devices with this issue - which makes writing inaccurate always - when competitors like Boox at least have a better ink-accuracy experience.

I'm happy to help test and work on the proposed suggestion.

In the meantime I'll start reading texts on eink hardware to see if I can better understand the hardware issue.

**Comment by hellerbarde** [comment]
I don't have one of these, but maybe someone with one could try it on the remarkable: https://en.wikipedia.org/wiki/Magnetic_field_viewing_film

**Comment by mshirey** [comment]
> @aogden41 look at this link, below. To me, this solved it all. Just swipe the magnet of your pen over the screen thoroughly. https://www.reddit.com/user/IoannisHere/comments/m83t6v/remarkable_2_tablet_screen_magnetization/

Thank you!  This should be in the manual!  I've read that this doesn't work for some, but for me, this worked PERFECTLY!

**Comment by stevenyoon79** [comment]
It was really frustrating because no one was talking about this issue.
Even when I spoke to remarkable HQ, the only response I received was that my device was defective.
They told me it might have been affected by magnetism because it was placed with a laptop?
It has been replaced twice, but of course the problem is still occurring.
Remarkable HQ can't possibly be unaware of this problem, but they haven't mentioned it at all and don't seem to have any intention of fixing it.
I even wrote this in the remarkable 2 usergroup on Facebook and it was immediately deleted.

**Comment by stevenyoon79** [comment]
it will help.
https://www.youtube.com/watch?v=4Tvu6tVzTA4&ab_channel=KidsHealthByDr.Tahir

demagnetizer.
